polyfills/WeakMap/polyfill.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @license
+ *
+ * Portions of this polyfill are a derivative work of the Polymer project, which requires the following licence notice:
+ *
+ * Copyright (c) 2014 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+ (function() {
+	var defineProperty = Object.defineProperty;
+	var counter = Date.now() % 1e9;
+
+	var WeakMap = function(data) {
+		var i, s;
+		this.name = '__st' + (Math.random() * 1e9 >>> 0) + (counter++ + '__');
+
+		// If data is iterable (indicated by presence of a forEach method), pre-populate the map
+		data && data.forEach && data.forEach(function (item) {
```

Comment:
Should we include a dependency on `Array.prototype.forEach`?  (Although technically 'data' can be the more generic `iterable`).